src/main/java/com/openshift/internal/restclient/model/Pod.java

```diff
@@ -61,10 +76,35 @@ public String getHost() {
 		}
 		return images;
 	}
+	
+	/**
+	 * The logic of the method is a copied from 'podStatus' function of
+	 * [app/scripts/filters/resources.js] of [openshift/origin-web-console]
+	 */
 
 	@Override
 	public String getStatus() {
-		return asString(POD_STATUS);
+		if (has(POD_DELETION_TIMESTAMP)) {
```

Comment:
@bdshadow yes but then the extensive logic that gets replicated across tools shows to me, that this belongs into the server, not into tools. Duplication will happen, but better have them in one single place than several times in all tools.